django/db/models/fields/related_lookups.py

```diff
@@ -81,6 +81,9 @@ def as_sql(self, compiler, connection):
                     AND)
             return root_constraint.as_sql(compiler, connection)
         else:
+            if getattr(self.rhs, '_forced_pk', False):
+                self.rhs.clear_select_clause()
+                self.rhs.add_fields([self.lhs.field.target_field.name], True)
```

Comment:
I'm not sure if this code is correct and in the best place. The `one_to_one` test failure is fixed by changing `self.lhs.field.target_field.name` to `self.lhs.field.name` but that causes failures in other tests.

```
ERROR: test_rel_pk_subquery (one_to_one.tests.OneToOneTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/one_to_one/tests.py", line 522, in test_rel_pk_subquery
    self.assertSequenceEqual(q2, [r])
 ...
  File "/home/tim/code/django/django/db/models/fields/related_lookups.py", line 86, in as_sql
    self.rhs.add_fields([self.lhs.field.target_field.name], True)
  File "/home/tim/code/django/django/db/models/sql/query.py", line 1659, in add_fields
    "Choices are: %s" % (name, ", ".join(names)))
django.core.exceptions.FieldError: Cannot resolve keyword 'id' into field.
 Choices are: favorites, place, place_id, serves_hot_dogs, serves_pizza, waiter